ts/src/pro/xt.ts

```diff
@@ -0,0 +1,1062 @@
+                //        error: null,
+                //        result: '3BC1D71D6CF96DA3458FC35B05B633351684511731128'
+                //    }
+                //
+                client.subscriptions['accessToken'] = this.safeValue (response, 'result');
+            } else {
+                const response = await this.privateSpotPostWsToken ();
+                //
+                //    {
+                //        "rc": 0,
+                //        "mc": "SUCCESS",
+                //        "ma": [],
+                //        "result": {
+                //            "accessToken": "eyJhbqGciOiJSUzI1NiJ9.eyJhY2NvdW50SWQiOiIyMTQ2Mjg1MzIyNTU5Iiwic3ViIjoibGh4dDRfMDAwMUBzbmFwbWFpbC5jYyIsInNjb3BlIjoiYXV0aCIsImlzcyI6Inh0LmNvbSIsImxhc3RBdXRoVGltZSI6MTY2MzgxMzY5MDk1NSwic2lnblR5cGUiOiJBSyIsInVzZXJOYW1lIjoibGh4dDRfMDAwMUBzbmFwbWFpbC5jYyIsImV4cCI6MTY2NjQwNTY5MCwiZGV2aWNlIjoidW5rbm93biIsInVzZXJJZCI6MjE0NjI4NTMyMjU1OX0.h3zJlJBQrK2x1HvUxsKivnn6PlSrSDXXXJ7WqHAYSrN2CG5XPTKc4zKnTVoYFbg6fTS0u1fT8wH7wXqcLWXX71vm0YuP8PCvdPAkUIq4-HyzltbPr5uDYd0UByx0FPQtq1exvsQGe7evXQuDXx3SEJXxEqUbq_DNlXPTq_JyScI",
+                //            "refreshToken": "eyJhbGciOiqJSUzI1NiJ9.eyJhY2NvdW50SWQiOiIyMTQ2Mjg1MzIyNTU5Iiwic3ViIjoibGh4dDRfMDAwMUBzbmFwbWFpbC5jYyIsInNjb3BlIjoicmVmcmVzaCIsImlzcyI6Inh0LmNvbSIsImxhc3RBdXRoVGltZSI6MTY2MzgxMzY5MDk1NSwic2lnblR5cGUiOiJBSyIsInVzZXJOYW1lIjoibGh4dDRfMDAwMUBzbmFwbWFpbC5jYyIsImV4cCI6MTY2NjQwNTY5MCwiZGV2aWNlIjoidW5rbm93biIsInVzZXJJZCI6MjE0NjI4NTMyMjU1OX0.Fs3YVm5YrEOzzYOSQYETSmt9iwxUHBovh2u73liv1hLUec683WGfktA_s28gMk4NCpZKFeQWFii623FvdfNoteXR0v1yZ2519uNvNndtuZICDdv3BQ4wzW1wIHZa1skxFfqvsDnGdXpjqu9UFSbtHwxprxeYfnxChNk4ssei430"
+                //        }
+                //    }
+                //
+                const result = this.safeValue (response, 'result');
+                client.subscriptions['accessToken'] = this.safeString (result, 'accessToken');
+            }
+        }
+        return client.subscriptions['accessToken'];
+    }
+
+    async subscribe (name: string, access: string, methodName: string, market: object = undefined, params = {}) {
+        /**
+         * @ignore
+         * @method
+         * @description Connects to a websocket channel
+         * @see https://doc.xt.com/#websocket_privaterequestFormat
+         * @see https://doc.xt.com/#futures_market_websocket_v2base
+         * @param {string} name name of the channel
+         * @param {string} access public or private
+         * @param {string} methodName the name of the CCXT class method
+         * @param {object|undefined} market CCXT market
+         * @param {object} params extra parameters specific to the xt api
+         * @returns {object} data from the websocket stream
+         */
+        await this.loadMarkets ();
+        const privateAccess = access === 'private';
+        let isContract = undefined;
+        [ isContract, params ] = this.handleIsContractAndParams (methodName, market, params);
```

Comment:
this is unnecessary, you can easily use `[ type, params ] = this.handleMarketTypeAndParams (methodName, market, params);`